imagekitio/url.py

```diff
@@ -211,3 +211,22 @@ def transformation_to_str(transformation):
             )
 
         return Default.CHAIN_TRANSFORM_DELIMITER.value.join(parsed_transforms)
+
+    @staticmethod
+    def encodeURI(url_str):
+        # https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI
+        if "?" in url_str:
+            # here, we have applied the quote(unquote(url)) function to handle the query string.
+            # This function ensures that characters in the query are properly encoded. While some characters are already encoded, others require encoding for correct processing.
+            encoded_url = quote(url_str.split('?')[0], safe='~@#$&()*!+=:;,?/\'')+"?"+quote(unquote(url_str.split('?')[1]), safe='~@#$&()*!+=:;?/\'')
+        else:
+            encoded_url = quote(url_str.split('?')[0], safe='~@#$&()*!+=:;,?/\'')
```

Comment:
Why we are splitting on `?` here in else condition?
Replace `url_str.split('?')[0]` with `url_str`.
Also make `'~@#$&()*!+=:;?/\''` a separate variable and use that for better code readability.

Reply:
addressed